src/dracut-cpio/src/main.rs

```diff
@@ -1810,4 +1797,40 @@ mod tests {
             assert!(gmd.rdev() == dmd.rdev());
         }
     }
+
+    #[test]
+    fn test_archive_all_single_threaded() {
+        // Each test_archive_ function changes its CWD and cargo test spawns
+        // multiple threads for concurrent testing by default, unless given
+        // --test-threads=1. Therefore, use a single wrapper test so that people
+        // don't get confused by multi-threaded "cargo test" failures.
+        // If someone finds a way make single-threaded the default for
+        // "cargo test" then we can drop this wrapper in future.
```

Comment:
I am not sure if that solution is the best one. I ran into a similar issue with 3cpio and the solution there is quite simple. I created:

```
    // Lock for tests that rely on / change the current directory
    pub static TEST_LOCK: std::sync::Mutex<u32> = std::sync::Mutex::new(0);
```

Then each test that relies on cwd gets this call as first step:

```
        let _lock = TEST_LOCK.lock().unwrap();
```